File: códigos en JAVA/ejercicio9ape6.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */

/**
 *
 * @author USUARIO
 */
    import java.util.LinkedList; // 
public class ejercicio9ape6 {

    public static void main(String[] args) {
        // En Java, LinkedList implementa las interfaces List y Deque
        LinkedList<String> turnos = new LinkedList<>();

        // Método para agregar al final
        turnos.addLast("T01 - Juan");
        turnos.addLast("T02 - Maria");
        turnos.addLast("T03 - Carlos");

        System.out.println("--- COLA DE ATENCION ---");
        System.out.println(turnos);

        // Método para obtener y remover el primer nodo
        String atendido = turnos.removeFirst();
        System.out.println("\nAtendiendo a: " + atendido);

        System.out.println("\n--- COLA ACTUALIZADA ---");
        System.out.println(turnos);
    }
}

File: códigos en C++/ejercicio9ape6.cpp
#include <iostream>
#include <list> //  Librería específica para listas doblemente enlazadas
#include <string>

using namespace std;

int main() {
    list<string> turnos;

    // Agregar turnos al final de la cola
    turnos.push_back("T01 - Juan"); //
    turnos.push_back("T02 - Maria");
    turnos.push_back("T03 - Carlos");

    cout << "--- COLA DE ATENCION ---" << endl;
    for(string turno : turnos) cout << turno << endl;

    // Atender (Eliminar el primer elemento)
    cout << "\nAtendiendo a: " << turnos.front() << endl;
    turnos.pop_front(); // Remueve de manera eficiente el nodo inicial

    cout << "\n--- COLA ACTUALIZADA ---" << endl;
    for(string turno : turnos) cout << turno << endl;

    return 0;
}
